### PR TITLE
Add Batch Predict to Documentation

### PR DESCRIPTION
**Pull request by simon-mo**
Add batch predict to:
1. tutorials
2. api doc

Update api doc to 0.3.0

image

image

image

### CODE REVIEW

**Comment by simon-mo** [comment]
Resolves https://github.com/ucbrise/clipper/issues/544

File: content/tutorials/batch_predict.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+date= 2017-09-12T23:45:24-07:00
+description = ""
+toc = false
+weight = 13
++++
+
+## Batch Prediction Interface
+
+### Request
+Since release 0.3.0, Clipper has a batch prediction inferface for query. Instead of sending:
```
Comment (RehanSD) [inline content/tutorials/batch_predict.md]: inferface -> interface

Comment (RehanSD) [inline content/tutorials/batch_predict.md]: inferface -> interface

File: content/tutorials/batch_predict.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+### Request
+Since release 0.3.0, Clipper has a batch prediction inferface for query. Instead of sending: 
+```json
+{ "input" := [double] | [int] | [byte] | [float] | string }
+```
+You can now send:
+```json
+{ "input_batch" := [[double] | [int] | [byte] | [float] | string] }
+```
+Clipper will return prediction for the batch in the original order.
```
Comment (RehanSD) [inline content/tutorials/batch_predict.md]: prediction*s*

File: content/tutorials/batch_predict.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	"batch_predictions":
+	[
+		{"query_id":2,"output":4.0,"default":false},
+		{"query_id":3,"output":5.0,"default":false}
+	]
+}
+```
+
+### Note
+If a request contain both `input` and `input_batch`, `input` will take precedence and `input_batch` will be ignored.
```
Comment (RehanSD) [inline content/tutorials/batch_predict.md]: contain*s*

File: content/tutorials/r-deployment.md [hunk 4]
```diff
@@ -15,6 +15,17 @@ functions. Given an API-compatible R function, Rclipper’s
 This model can then be deployed to Clipper via the [Python
 clipper\_admin package](https://pypi.python.org/pypi/clipper_admin).
 
+Installation
+------------
+You can install rclipper package by cloning Clipper repository
```
Comment (RehanSD) [inline content/tutorials/r-deployment.md]: install *the* rclipper package

Comment (RehanSD) [inline content/tutorials/r-deployment.md]: cloning *the* Clipper repository

File: content/tutorials/r-deployment.md [hunk 5]
```diff
@@ -15,6 +15,17 @@ functions. Given an API-compatible R function, Rclipper’s
...
+and run `R CMD INSTALL`:
```
Comment (RehanSD) [inline content/tutorials/r-deployment.md]: run*ning*

**Review by RehanSD — CHANGES_REQUESTED** [review]

File: content/tutorials/batch_predict.md (hunk 1 above)
Comment (RehanSD) [inline content/tutorials/batch_predict.md]: Starting

**Comment by covix** [comment]
I just spent a few hours trying to understand how the batch predictions works (together with some strange query behaviour regarding the `input` parameter).

I believe that having the doc updated could help people.
Is this ok to be merged? If not, can I do something to help?

**Comment by rkooo567** [comment]
@simon-mo Can I have an access to clipper-websites? Also, what do you think about @covix's comment? I am not sure why this PR is not merged? 

@covix I will get back to you within a few days to answer about your question!

**Comment by rkooo567** [comment]
@covix I haven't heard back from @simon-mo. I think it is okay to create a new PR for this. We will appreciate your contribution!

**Pull request closed without merging**